test-alerts.cpp

```diff
@@ -5,4 +5,28 @@
 
 TEST_CASE("infers the breach according to limits") {
   REQUIRE(inferBreach(12, 20, 30) == TOO_LOW);
+  REQUIRE(inferBreach(42, 20, 30) == TOO_HIGH);
+  REQUIRE(inferBreach(25, 20, 30) == NORMAL);
+}
+
+TEST_CASE("Alerts target based on breach type") {
+  alertTarget(TO_CONTROLLER, TOO_LOW);
```

Comment:
This test is only calling alertTarget, not testing the intended functionality.
Try dependency injection to separate the interface from the implementation and the stub. Then use it to test the content of the messages being sent out.

Reply:
updated the test case and did the code changes to test